sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -159,13 +159,14 @@ private static HttpMessageHandler CreateDefaultHandler(HttpPipelineTransportOpti
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
             {
+                // UseCookies is not supported on "browser"
                 return new HttpClientHandler();
             }
 
 #if NETCOREAPP
-            return ApplyOptionsToHandler(new SocketsHttpHandler { AllowAutoRedirect = false }, options);
+            return ApplyOptionsToHandler(new SocketsHttpHandler { AllowAutoRedirect = false, UseCookies = UseCookies() }, options);
```

Comment:
Do we want to have this apply to the shared instance or make it client specific by making it part of https://github.com/Azure/azure-sdk-for-net/blob/f42c7461b95912aca826b08957d8a974594ad61e/sdk/core/Azure.Core/src/Pipeline/HttpPipelineTransportOptions.cs

Reply:
I think for now we can just have it apply to the shared instance. If someone creates their own HttpClientHandler, they can set the UseCookies property to whatever they want. I don't think we can add it to the options bag until we support it in HttpWebRequest as well, and there is no reason to do so, at this point.

Reply:
Makes sense if this is only for back compat and not some scenario that requires enabling `UseCookies`. If we did have a scenario though, the advantage of adding to the options is that it will create a transport instance that is configured with the Core defaults in every other way, rather than starting from a purely default instance of the handler.

Reply:
Yeah this is strictly for back compat.

Reply:
>I think for now we can just have it apply to the shared instance. If someone creates their own HttpClientHandler, they can set the UseCookies property to whatever they want

If this feature is for back compat and to allow easily switching the old behavior back we need to fully switch cookies on, not only in shared case.

Reply:
cookies are on by default when using the constructor that takes a message handler, or maybe there is another scenario?

Reply:
cookies are configured in the HttpMessageHandler when using either the HttpClient ctor or the HttpMessageHandler ctor. Customers can set the cookies however they want here, so I don't think we should override their setting.

Reply:
I should clarify, this change isn't only applied to the shared instance. It is applied to any instance that doesn't bring its own HttpClient/HttpMessageHandler.